### ISSUE TITLE
Images do not show

### ISSUE THREAD

**Issue by Aida-Enna** [discordapp/discord-rpc#21]
I'm using the example code provided in a C# project and everything is working fine *except* for the images, both large and small. I've uploaded them to the dashboard and am using the corresponding key (the name I gave the image, that shows up next to it post upload). I've yet to get any of the images to show.

`
            presence.largeImageKey = "lobby";
`

![image](https://user-images.githubusercontent.com/5483362/32672103-deb54816-c617-11e7-83e1-4b58ba40a585.png)

![image](https://user-images.githubusercontent.com/5483362/32672186-1f43936a-c618-11e7-88a6-e3113f5a53d7.png)

**Comment by Chaosvex** [comment]
Exactly the same issue here and as https://github.com/discordapp/discord-rpc/issues/14.

**Comment by Aida-Enna** [comment]
Restarting both discord clients and the program didn't fix the issue, unfortunately.

**Comment by Chaosvex** [comment]
The pictures eventually decided to join the party but freshly uploaded pictures still don't work. I'm assuming there's a huge delay (6-8 hours?) on uploading pictures and actually being able to make use of them.

**Comment by YaLTeR** [comment]
Same issue. I guess I'll have to wait.

**Comment by night** [comment]
This should be fixed now.

If you're still having issues you probably have old cached data. Follow these steps to clear it:

1. Open Discord console with ctrl + shift + I (cmd + shift + I on macOS)
2. Go to the Network tab
3. Toggle "Disable cache"
4. Refresh the app with ctrl + R (cmd + R on macOS).
5. Send rich presence containing an asset to confirm it's fixed

**Comment by Aida-Enna** [comment]
Just came back to this after a long break - My images are still not showing, even after doing the instructions provided by @night. Can anyone else confirm that this is fixed for them? =/

**Comment by YaLTeR** [comment]
Yeah, the images showed up for me after a couple hours. No idea if anything was done about the long wait since then.

**Comment by msciotti** [comment]
Did you disable the cache as previously instructed? And is your app up to date?

**Comment by Aida-Enna** [comment]
Yep, I disabled the cache and I tried the latest versions of Discord and Canary. Would the pictures need to be re-uploaded to work?

**Comment by msciotti** [comment]
They _shouldn't_, but as with all things programming related, it never hurts to try it lol

**Comment by DeepDoge** [comment]
> The pictures eventually decided to join the party but freshly uploaded pictures still don't work. I'm assuming there's a huge delay (6-8 hours?) on uploading pictures and actually being able to make use of them.

i know this is an old thread but i need to share this because it was a pain for me

i have a personal youtube and soundcloud rpc and it uploads thumbnails automatically
i had the same problem

then realized that the page where you upload assets is using 
( https://discordapp.com/api/oauth2/applications/{your app id}/assets )
to get the assets list.
it works instantly 
after you upload the asset you can instantly see it on the list

BUT discord client uses
( https://discordapp.com/api/v6/oauth2/applications/{your app id}/assets )
to get the assets list
and it doesn't work instantly all the time 
im not sure why

im using both.
using the first one to get a list of already uploaded assets
and using the second one to check if the asset is ready to show on rpc

**Comment by Lachee** [comment]
uploading of images automatically is technically not supported (since its not documented anywhere) and teeechnically against the ToS. This upload delay is probably a one of the many reasons they havn't opened this endpoint to the public. 

As for the V6 part of the URL, that shouldnt change anything as the default API is version 6 at the moment. The reason you can see it instantly is if you are uploading via the page, it has some JS to insert a new element into the list for you.

In short, with the intended use of Rich Presence this is a non-issue sadly.

**Comment by DeepDoge** [comment]
Well, v6 changing things. and as i said I'm not sure why. And JS has nothing to the with that.
I'm talking about direct HTTP requests to the API.

When you update the RPC asset key, discord client asks for an array of assets then finds the id of the asset and uses that id to create that URL ( https://cdn.discordapp.com/app-assets/{app_id}/{asset_id}.png ).

The upload request also returns you the generated id of the asset too.
So after uploading the asset, first, I tried to check if its useable by sending a request to URL above.
But it works instantly too just like ( https://discordapp.com/api/oauth2/applications/{app_id}/assets ).
(reminder I'm talking about HTTP requests)

So after uploading the picture and discord told you its "DONE", you can instantly access it from both of the URLs above. I checked it many times. The only thing that has a delay on it sometimes is v6. 
Discord can't show the image because it can't get the id of the asset key because it can't find it on the array. So it can't create the png URL. 
I have a few guesses but I'm not sure why it delays.

**Comment by msciotti** [comment]
Hey @DeepDoge - would you mind opening a separate issue for this? Just so we aren't spamming OP with GitHub notifications.

**Comment by brr53** [comment]
I am still having this issue.

**Comment by ritiksharmarj** [comment]
<img width="1348" alt="Screenshot 2023-12-06 at 12 52 36 PM" src="https://github.com/discord/discord-rpc/assets/54701022/08998da6-4118-47fc-bfe1-d73d0e700edb">

Is anyone still facing this issue because I am? What's the solution
